src/pint/models/absolute_phase.py

```diff
@@ -108,16 +108,28 @@ def get_TZR_toa(self, toas):
         #     ephem=toas.ephem,
         #     planets=toas.planets,
         # )
-        tz = toa.get_TOAs_array(
-            (self.TZRMJD.quantity.jd1 - 2400000.5, self.TZRMJD.quantity.jd2),
-            obs=self.TZRSITE.value,
-            freqs=self.TZRFRQ.quantity,
-            include_bipm=clkc_info["include_bipm"],
-            include_gps=clkc_info["include_gps"],
-            ephem=toas.ephem,
-            planets=toas.planets,
-            tzr=True,
-        )
+        if self.TZRSITE.value != "@" and self.TZRSITE.value.lower() != "bat":
+            tz = toa.get_TOAs_array(
+                (self.TZRMJD.quantity.jd1 - 2400000.5, self.TZRMJD.quantity.jd2),
+                obs=self.TZRSITE.value,
+                freqs=self.TZRFRQ.quantity,
+                include_bipm=clkc_info["include_bipm"],
+                include_gps=clkc_info["include_gps"],
+                ephem=toas.ephem,
+                planets=toas.planets,
+                tzr=True,
+            )
+        else:
+            tz = toa.get_TOAs_array(
+                (self.TZRMJD.quantity.jd1 - 2400000.5, self.TZRMJD.quantity.jd2),
+                obs=self.TZRSITE.value,
+                freqs=self.TZRFRQ.quantity,
+                include_bipm=False,
+                include_gps=False,
+                ephem=toas.ephem,
+                planets=toas.planets,
+                tzr=True,
+            )
         log.debug("Done with TZR_toa")
         self.tz_cache = tz
```

Comment:
@abhisrkckl  I think in here `self.tz_clck_info` should also reflect the fact that if TZRMJD is barycentered, no corrections have been applied. What do you think?

Reply:
Sorry, it would be line 124/136, it is just outside the commit and cannot comment directly on it.

Reply:
Please see my comment above. I think if we fix the bug in `get_observatory`, this should get fixed automatically.